Source/ACE/Entity/Player.cs

```diff
@@ -194,6 +194,43 @@ public void HandleActionLearnSpell(uint spellId)
             learnSpellChain.EnqueueChain();
         }
 
+        /// <summary>
+        /// This method implements player spell bar management for - adding a spell to a specific spell bar (0 based) at a specific slot (0 based).
+        /// </summary>
+        /// <param name="spellId"></param>
+        /// <param name="spellBarPositionId"></param>
+        /// <param name="spellBarId"></param>
+        public void HandleActionAddSpellToSpellBar(uint spellId, uint spellBarPositionId, uint spellBarId)
+        {
+            // The spell bar magic happens here. First, let's mind our race conditions....
+            // No need for action chain here - internal to us, no other object can change us.
```

Comment:
This can be read and written at the same time (You can learn a spell while deleting/casting/moving the spellbar).  Even though its internal to us, its concurrent.  In general don't bother trying to elide ActionChains, your more likely to introduce bugs that are difficult to debug later on.